src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const service = require('../services/expenses.services');
+const { getById: getUserById } = require('../services/users.services');
+const {
+  OK, BAD_REQUEST, NOT_EXIST, CREATED, SUCCES_NO_CONTENT,
+} = require('../statusCodes');
+
+const getAllExpenses = (req, res) => {
+  const {
+    userId,
+    from,
+    to,
+    categories,
+  } = req.query;
+
+  let expenses = service.getAll();
+
+  if (userId) {
+    expenses = expenses.filter(expense => expense.userId === Number(userId));
+  }
+
+  if (from) {
+    expenses = expenses.filter(expense =>
+      new Date(expense.spentAt).valueOf() >= new Date(from).valueOf());
+  }
+
+  if (to) {
+    expenses = expenses.filter(expense =>
+      new Date(expense.spentAt).valueOf() <= new Date(to).valueOf());
+  }
+
+  if (categories) {
+    expenses = expenses.filter(
+      expense => categories.includes(expense.category)
```

Comment:
if there are no `categories` in search params we will get `undefined`
if only 1 `categories` we will get just value (....?categories=1)
And if we have many categories we will get array of values (....?categories=1&categories=2&categories=3)

![image](https://github.com/mate-academy/node_accounting-app-v2/assets/86789161/ea010564-92a2-4116-9fcf-a66739d6d01f)